packages/react/src/components/Forms/FileUploader/index.tsx

```diff
@@ -0,0 +1,347 @@
+import React, {
+  forwardRef,
+  PropsWithChildren,
+  ReactNode,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import classNames from "classnames";
+import { Field, FieldProps, FieldState } from ":/components/Forms/Field";
+import { Button } from ":/components/Button";
+import { useCunningham } from ":/components/Provider";
+import {
+  formatBytes,
+  replaceInputFilters,
+} from ":/components/Forms/FileUploader/utils";
+import { InputRefType } from ":/components/Forms/Input";
+import { Loader } from ":/components/Loader";
+
+export interface FileUploaderProps extends Omit<FieldProps, "state"> {
+  state?: FieldState | "uploading" | undefined;
+  multiple?: boolean;
+  icon?: ReactNode;
+  successIcon?: ReactNode;
+  deleteIcon?: ReactNode;
+  fileSelectedIcon?: ReactNode;
+  uploadingIcon?: ReactNode;
+  animateIcon?: boolean;
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+  name?: string;
+  bigText?: string;
+  onChange?: (event: { target: { value: File[] } }) => void;
+  // This is only here for storybook. It cannot be used in real conditions.
+  fakeDefaultFiles?: File[];
+}
+
+interface DropZoneProps extends FileUploaderProps, PropsWithChildren {
+  files: File[];
+  onFilesChange: (files: File[]) => void;
+}
+
+export interface FileUploaderRefType extends InputRefType {
+  reset: () => void;
+}
+
+const DropZone = forwardRef<FileUploaderRefType, DropZoneProps>(
+  (
+    {
+      multiple,
+      name,
+      inputProps,
+      state,
+      icon,
+      animateIcon,
+      successIcon,
+      uploadingIcon,
+      text,
+      bigText,
+      files,
+      onFilesChange,
+      onChange,
+      children,
+    }: DropZoneProps,
+    ref
+  ) => {
+    const [dragActive, setDragActive] = useState(false);
+    const container = useRef<HTMLLabelElement>(null);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const { t } = useCunningham();
+
+    useImperativeHandle(ref, () => ({
+      get input() {
+        return inputRef.current;
+      },
+      reset() {
+        onFilesChange([]);
+      },
+    }));
+
+    useEffect(() => {
+      if (!inputRef.current) {
+        return;
+      }
+      replaceInputFilters(inputRef.current, files);
+    }, [files]);
+
+    useEffect(() => {
+      onChange?.({ target: { value: files ?? [] } });
+    }, [files]);
+
+    const renderIcon = () => {
+      if (state === "success") {
+        return successIcon ?? <span className="material-icons">done</span>;
+      }
+      if (state === "uploading") {
+        return (
+          uploadingIcon ?? (
+            <Loader
+              size="small"
+              aria-label={t("components.forms.file_uploader.uploading")}
+            />
+          )
```

Comment:
In order to ensure a11y, what do you to enforce aria-label prop.
```suggestion
React.cloneElement(
  uploadingIcon ?? <Loader size="small" />,
  {"aria-label": t("components.forms.file_uploader.uploading")}
)
```

Reply:
Good catch!!